### PR TITLE
Merge feat esdt prefix into sovereign 7 oct 2024

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- 
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: process/coordinator/sovereignChainProcess.go [hunk 1]
```diff
@@ -63,6 +63,16 @@ func (sctc *sovereignChainTransactionCoordinator) CreateMbsAndProcessCrossShardT
...
 	shouldSkipShard := make(map[uint32]bool)
 
+	headerHash, err := core.CalculateHash(sctc.marshalizer, sctc.hasher, hdr)
+	if err != nil {
+		log.Warn("transactionCoordinator.CreateMbsAndProcessCrossShardTransactionsDstMe CalculateHash error",
+			"error", err)
+
+		// we return the nil error here as to allow the proposer execute as much as it can, even if it ends up in a
+		// totally unlikely situation in which it can not marshall a block.
+		return createMBDestMeExecutionInfo.miniBlocks, createMBDestMeExecutionInfo.numTxAdded, false, nil
```
Comment (sasurobert) [inline process/coordinator/sovereignChainProcess.go:73]: return error here. this is critical error.

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: process/coordinator/sovereignChainProcess.go (hunk 1 above)
Comment (sasurobert) [reply process/coordinator/sovereignChainProcess.go:73]: yes. return error.

**Review by axenteoctavian — DISMISSED** [review]
Looks ok. 
- I tested some transactions with tokens with prefix on this branch and it works 
- I will create another PR with some updates/improvements for the sovereign chain simulator tests

File: process/coordinator/sovereignChainProcess.go (hunk 1 above)
Comment (mariusmihaic) [reply process/coordinator/sovereignChainProcess.go]: Ok, returned error

**Review by sasurobert — APPROVED** [review]

**Review by axenteoctavian — APPROVED** [review]

**Pull request merged**